pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -49,18 +50,26 @@ type DiffListChange struct {
 	Diff *diff.Result
 }
 
-func Diff(old, new Manifest, opts ...diff.Option) (*diff.Result, error) {
+func Diff(old, new Manifest, logger *zap.Logger, opts ...diff.Option) (*diff.Result, error) {
 	if old.Key.IsSecret() && new.Key.IsSecret() {
 		var err error
 		new.u, err = normalizeNewSecret(old.u, new.u)
 		if err != nil {
 			return nil, err
 		}
 	}
+
+	normalized, err := remarshal(old.u)
+	if err != nil {
+		logger.Info("Unable to remarshal Kubernetes manifest, the raw data will be used to calculate the diff", zap.Error(err))
+	} else {
+		old.u = normalized
+	}
+
 	return diff.DiffUnstructureds(*old.u, *new.u, opts...)
```

Comment:
`old.u = normalized` will override the underlying data of manifest which could be stored in cache or somewhere to be shared. So I think they should only be read-only to avoid unexpected bugs.

```suggestion

	normalized, err := remarshal(old.u)
	if err != nil {
		logger.Info("Unable to remarshal Kubernetes manifest, the raw data will be used to calculate the diff", zap.Error(err))
	        return diff.DiffUnstructureds(*old.u, *new.u, opts...)
	}

	return diff.DiffUnstructureds(*normalized, *new.u, opts...)
```

Reply:
I'm sorry. I thought I had applied this change, but I guess I hadn't.
btw, I think `old.u = normalized` does not override the underlying data (`*old.u = *normalized` does). So I don't think this change will make any difference with respect to behavior. However, it might be better to change it in terms of readability. How do you think?

Reply:
@Hosshii You are totally right. Thanks for your explanation. Go ahead with your idea.